ivi/rohdeschwarz/rohdeschwarzBaseScope.py

```diff
@@ -606,15 +608,23 @@ def _set_channel_show_label(self, index, value):
     def _get_channel_enabled(self, index):
         index = ivi.get_index(self._channel_name, index)
         if not self._driver_operation_simulate and not self._get_cache_valid(index=index):
-            self._channel_enabled[index] = bool(int(self._ask("%s:state?" % self._channel_name[index])))
-            self._set_cache_valid(index=index)
+            if index < self._analog_channel_count:
+                self._channel_enabled[index] = bool(int(self._ask("%s:state?" % self._channel_name[index])))
+                self._set_cache_valid(index=index)
+            else:
+                self._channel_enabled[index] = bool(int(self._ask("Logic1:state?")))
+                self._set_cache_valid(index=index)
         return self._channel_enabled[index]
 
     def _set_channel_enabled(self, index, value):
         value = bool(value)
         index = ivi.get_index(self._channel_name, index)
         if not self._driver_operation_simulate:
-            self._write("%s:state %d" % (self._channel_name[index], int(value)))
+            if index < self._analog_channel_count:
+                self._write("%s:state %d" % (self._channel_name[index], int(value)))
+            else:
+                state = ['OFF', 'ON']
+                self._write("Logic1:state %s" % (state[int(value)]))
```

Comment:
Same as above. Would it be better to handle digital channels instead of the whole bus? Just a thought.